### PR TITLE
Allow to customize form titles.

### PR DESCRIPTION
**Pull request by almet**
Use a registry rather than passing yet another prop to children.

r? @n1k0

### CODE REVIEW

File: test/index_test.js [hunk 1]
```diff
@@ -541,6 +541,16 @@ describe("Form", () => {
         .eql("my object");
     });
 
+    it("should render a customized title", () => {
+      const CustomTitleField = function(props) {
+        return (<div id="custom">{ props.title }</div>);
+      };
```
Comment (n1k0) [inline test/index_test.js]: Nit: 

```js
const CustomTitleField = ({title}) => <div id="custom">{title}</div>;
```

File: src/components/fields/ObjectField.js [hunk 2]
```diff
@@ -81,7 +82,7 @@ if (process.env.NODE_ENV !== "production") {
     onChange: PropTypes.func.isRequired,
     formData: PropTypes.object,
     required: PropTypes.bool,
-    SchemaField: PropTypes.func.isRequired,
+    registry: PropTypes.object,
```
Comment (n1k0) [inline src/components/fields/ObjectField.js]: Would be great if we could describe a little more what's expected as a data structure here, eg. expected keys and their type.

File: src/components/fields/ObjectField.js [hunk 3]
```diff
@@ -39,7 +39,8 @@ class ObjectField extends Component {
   render() {
     const {schema, uiSchema, name} = this.props;
     const title = schema.title || name;
-    const SchemaField = this.props.SchemaField;
+    const SchemaField = this.props.registry.SchemaField;
+    const TitleField = this.props.registry.TitleField;
```
Comment (n1k0) [inline src/components/fields/ObjectField.js]: Nit:

```js
const {SchemaField, TitleField} = this.props.registry;
```

File: src/components/fields/TitleField.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+function TitleField(props) {
+  return <legend>{props.title}</legend>;
+}
```
Comment (n1k0) [inline src/components/fields/TitleField.js]: Would be great to have propTypes defined as for other components.

**Comment by n1k0** [comment]
Note: This will break BC for consumers of the current API, because you now have to pass a registry object instead of a SchemaField. We need to document this change in the release notes with a big red warning.

**Comment by almet** [comment]
> Note: This will break BC for consumers of the current API, because you now have to pass a registry object instead of a SchemaField. We need to document this change in the release notes with a big red warning.
This won't, you still pass `SchemaField`, the registry is only exposed internally.

**Comment by n1k0** [comment]
Ah right, cool. So we only need to update the docs and we're good to go with nits addressed.

**Comment by almet** [comment]
Docs updated.

**Comment by n1k0** [comment]
LGTM, r+

**Comment by n1k0** [comment]
Could you take care of releasing v0.11.0?

**Comment by almet** [comment]
Did this:

```bash
npm run dist
vim package.json
npm publish
git commit -am "Bump v0.11.0"
git tag v0.11.0
git push
```

**Comment by n1k0** [comment]
No `npm run build`? Did you test installing the package after the release?

**Comment by almet** [comment]
In the `package.json` file, I see `"dist": "npm run build:lib && npm run build:dist && npm run build:css",`.

Despite that, I still need to run `npm run build`?

**Comment by n1k0** [comment]
Ah no it's fine then :)

**Pull request merged**